src/net/sourceforge/kolmafia/request/MallPurchaseRequest.java

```diff
@@ -28,7 +28,7 @@
 public class MallPurchaseRequest extends PurchaseRequest {
   private static final Pattern YIELD_PATTERN =
       Pattern.compile(
-          "You may only buy ([\\d,]+) of this item per day from this store\\.You have already purchased ([\\d,]+)");
+          "You may only buy ([\\d,]+) of this item per day from this store\\. You have already purchased ([\\d,]+)");
```

Comment:
Captured by me: 
You may only buy 3 of this item per day from this store. You have already purchased 3 in the last 24 hours.

Pattern is valid.